cfme/common/vm.py

```diff
@@ -342,6 +363,53 @@ def get_detail(self, properties=None, icon_href=False):
         else:
             return InfoBlock.text(*properties)
 
+    def open_console(self, console='VM Console', invokes_alert=False, cancel=False):
+        """
+        Initiates the opening of one of the console types supported by the Access
+        button.   Presently we only support VM Console, which is the HTML5 Console.
+        In case of VMware provider it could be VMRC, VNC/HTML5, WebMKS, but we only
+        support VNC/HTML5.
+        Possible values for 'console' could be 'VM Console' and 'Web Console', but Web
+        Console is not supported as well.
+
+        Args:
+            console       - one of the supported console types given by the Access button.
+            invokes_alert - If the particular console will invoke a CFME popup/alert
+                            setting this to true will handle this.
+            cancel        - Allows one to cancel the operation if the pope/alert occurs.
+        """
+        # TODO: implement vmrc vm console
+        if console not in ['VM Console']:
+            raise NotImplementedError('Not supported console type: {}'.format(console))
+
+        navigate_to(self, 'Details')
+
+        # Click console button given by type
+        access_btn(console, invokes_alert=invokes_alert)
+
+        # Some access types in the future may require a CFME popup be
+        # handled.  The one that is supported now does not require this,
+        # and hence the default is not do this.
+        if invokes_alert:
+            sel.handle_alert(cancel)
+
+        # Get the consoles window handle, and then create a VMConsole object, and store
+        # the VMConsole object aside.
+        console_handle = self.console_handle
+        appliance_handle = self.appliance.browser.widgetastic.window_handle
+        logger.info("Creating VMConsole:")
+        logger.info("   appliance_handle: {}".format(appliance_handle))
+        logger.info("     console_handle: {}".format(console_handle))
+        logger.info("               name: {}".format(self.name))
+        vm_console = VMConsole(
```

Comment:
**QUESTION:** Why not just ```self.vm_console = vm_console``` also, as it can be so easily disrupted, I wonder if it's a good idea to store it on the vm object....